### PR TITLE
V2: Fix AUPRC + Add CLI tests for all available classification metrics

### PR DESCRIPTION
**Pull request by hwpang**
## Description
Found some bugs while running chemprop v2 benchmark related to AUPRC. Fixed it. While I'm at it, I also added more CLI tests for all available classification metrics to avoid any similar bugs.

## Example / Current workflow
n/a

## Bugfix / Desired workflow
- The functions used to calculate AUPRC are not available in torchmetrics. Fixed to use correct function name and import auc from sklearn.

- Current CLI test only tests AUCROC. I added all available metrics to the CLI test.

- Many of the classification metrics requires `task` to sepcify whether it's binary, multiclass, or multilabel. Added a `TaskedMixin` to supply the `task` argument.

## Questions
n/a

## Relevant issues
n/a

## Checklist
- [x] linted with flake8?
- [x] (if appropriate) unit tests added?

### CODE REVIEW

**Comment by KnathanM** [comment]
I saw the same issue and opened an issue #794. Then I immediately saw you already opened a PR, haha.

In v1 they used precision_recall_curve and then auc. https://github.com/chemprop/chemprop/blob/02c760216ec03488b69b41abe184589b9c996d74/chemprop/train/metrics.py#L125 What if we did the same with torchmetrics precision_recall_curve?

File: chemprop/nn/metrics.py [hunk 1]
```diff
@@ -144,33 +148,57 @@ def _calc_unreduced_loss(self, preds, targets, mask, *args) -> Tensor:
...
     minimize = False
 
+    def __init__(self, task: str, **kwargs):
+        super().__init__(**kwargs)
+        self.task = task
+
     def forward(self, preds: Tensor, targets: Tensor, *args, **kwargs):
-        p, r, _ = F.precision_recall(preds, targets.long())
+        p, r, _ = F.precision_recall_curve(preds, targets.long(), task=self.task)
+        return auc(r, p)
```
Comment (KnathanM) [inline chemprop/nn/metrics.py:164]: torchmetrics uses `torch.trapz(p, r) * -1` to calculate the auc.
See [here](https://github.com/Lightning-AI/torchmetrics/blob/be3970810a38753bfe53b8b162743de63bd6b179/src/torchmetrics/classification/precision_recall_curve.py#L219) and [here](https://github.com/Lightning-AI/torchmetrics/blob/master/src/torchmetrics/utilities/compute.py#L88)

**Comment by hwpang** [comment]
@KnathanM This is ready for another review & merge!

**Review by KnathanM — APPROVED** [review]
My PR #779 makes it difficult to have task as an argument to the metrics so I forced task to be binary and renamed the metrics. We will probably want to either add multiclass versions of these metrics or find a better way, but I think that can be handled in v2.1 and I opened an issue for that #817.

**Pull request merged**